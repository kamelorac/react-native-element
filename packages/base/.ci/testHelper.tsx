import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
