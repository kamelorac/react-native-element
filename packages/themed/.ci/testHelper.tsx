import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
