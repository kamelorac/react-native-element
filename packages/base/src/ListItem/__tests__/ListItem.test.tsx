import React from 'react';
import { ListItem } from '..';
import { renderWithWrapper } from '../../../.ci/testHelper';
import { Avatar, Icon } from '../..';
import { TextInput } from 'react-native';

describe('ListItem component', () => {
  it('should match snapshot', () => {
    const component = renderWithWrapper(<ListItem />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render with avatar', () => {
    const { wrapper } = renderWithWrapper(
      <ListItem containerStyle={{ backgroundColor: 'peru' }}>
        <Avatar source={{ uri: 'avatar_uri' }} />
      </ListItem>
    );
    expect(wrapper.findByType(Avatar).props).toMatchObject({
      source: { uri: 'avatar_uri' },
    });
  });

  it('should render with left icon', () => {
    const { wrapper } = renderWithWrapper(
      <ListItem>
        <Icon name="wifi" type="font-awesome" color="red" size={20} />
      </ListItem>
    );
    expect(wrapper.findByType(Icon).props).toMatchObject({
      name: 'wifi',
      color: 'red',
    });
  });

  it('should render with topDivider and bottomDivider', () => {
    const { wrapper } = renderWithWrapper(
      <ListItem topDivider={true} bottomDivider={true} />,
      'RNE__LISTITEM__padView'
    );
    expect(wrapper.props.style).toMatchObject({
      borderTopWidth: 0.5,
      borderBottomWidth: 0.5,
    });
  });

  it('should render with disabled true and styles', () => {
    const { wrapper } = renderWithWrapper(
      <ListItem
        disabled={true}
        containerStyle={{ padding: 10 }}
        disabledStyle={{ backgroundColor: 'gray' }}
      />,
      'RNE__LISTITEM__padView'
    );
    expect(wrapper.props.style).toMatchObject({
      backgroundColor: 'gray',
    });
  });

  it('should render with title and subtitle', () => {
    const component = renderWithWrapper(
      <ListItem>
        <ListItem.Content>
          <ListItem.Title numberOfLines={5} style={{ backgroundColor: 'peru' }}>
            title test
          </ListItem.Title>
          <ListItem.Subtitle style={{ backgroundColor: 'peru' }}>
            title test
          </ListItem.Subtitle>
        </ListItem.Content>
        <ListItem.Content right>
          <ListItem.Title right>title</ListItem.Title>
        </ListItem.Content>
      </ListItem>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should warn the user when using linearGradient without it installed', () => {
    console.warn = jest.fn();
    renderWithWrapper(
      <ListItem
        linearGradientProps={{ colors: ['#4c669f', '#3b5998', '#192f6a'] }}
      />
    );
    expect((console.warn as jest.Mock).mock.calls[0][0]).toBe(
      "You need to pass a ViewComponent to use linearGradientProps !\nExample: ViewComponent={require('react-native-linear-gradient')}"
    );
  });

  it('should render with input', () => {
    const onChange = jest.mock;
    const { wrapper } = renderWithWrapper(
      <ListItem>
        <ListItem.Input placeholder="Enter Text" onChangeText={onChange} />
      </ListItem>
    );
    const input = wrapper.findByType(TextInput);
    expect(input.props.placeholder).toBe('Enter Text');
  });
  it('should render accordion', () => {
    const { wrapper } = renderWithWrapper(
      <ListItem.Accordion
        isExpanded={false}
        icon={<Icon name="place" />}
        noRotation
      ></ListItem.Accordion>,
      'RNE__ListItem__Accordion__Icon'
    );
    expect(wrapper.props.style).toMatchObject({
      transform: [{ rotate: '0deg' }],
    });
  });
});
