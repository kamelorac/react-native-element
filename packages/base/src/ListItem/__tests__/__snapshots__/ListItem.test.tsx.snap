// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    />
  </View>
</View>
`;

exports[`ListItem component should render with title and subtitle 1`] = `
<View
  testID="wrapper"
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
      testID="RNE__LISTITEM__padView"
    >
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityRole="text"
          numberOfLines={5}
          style={
            {
              "backgroundColor": "peru",
              "color": "#242424",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          title test
        </Text>
        <Text
          accessibilityRole="text"
          style={
            {
              "backgroundColor": "peru",
              "color": "#242424",
              "fontSize": 15,
            }
          }
          testID="listItemTitle"
        >
          title test
        </Text>
      </View>
      <View
        style={
          {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          [
            {
              "alignItems": "flex-end",
              "flex": 0.5,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          accessibilityRole="text"
          style={
            {
              "backgroundColor": "transparent",
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
        >
          title
        </Text>
      </View>
    </View>
  </View>
</View>
`;
